vector/src/main/scala/geotrellis/vector/package.scala

```diff
@@ -93,4 +93,56 @@ package object vector extends SeqMethods {
 
   implicit def seqGeometryToGeometryCollection(gs: Seq[Geometry]): GeometryCollection =
     GeometryCollection(gs)
+
+  /* An extension class - when in scope, these methods are available from Geometry objects.
+   *
+   * The algorithms herein are all implemented in JTS, but the wrapper methods
+   * here make it straightforward to call them with geotrellis.vector classes.
+   */
+  implicit class withGeometryMethods[G <: Geometry](val self: G) extends MethodExtensions[G] {
```

Comment:
To fit with the pattern, we would have this extend traits instead of defining in the implicit class.

e.g.

```scala
package geotrellis.vector.dissolve

DissolveMethods[G <: Geometry] extends MethodExtensions[G] {
  /**
     * dissolve composes all the unique linestrings and points in the input
     * geometry and produces a linear geometry with no repeated components.
     */
    def dissolve(): LineOneDimensionUnionResult =
      // Could just inline, get rid of Dissolve object...the object separation technique
      // is mostly for spark's sake, where we need to ensure that the implicit
      // class is not serialized.
      Dissolve(self) 
}
```

and here in `package.scala`,

```scala
 implicit class withGeometryMethods[G <: Geometry](val self: G)
    extends MethodExtensions[G]
    with DissolveMethods[G]
```